module_4/src/com/epam/aggregationandcomposition/entity/car/Car.java

```diff
@@ -0,0 +1,128 @@
+package com.epam.aggregationandcomposition.entity.car;
+
+import com.epam.aggregationandcomposition.exceptions.CarLogicalException;
+import com.epam.aggregationandcomposition.exceptions.CarTechnicalException;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class Car {
+
+    private static final float FULL_TANK = 1.0f;
+    private static final float FUEL_AMOUNT_TO_ONE_RIDE = 0.2f;
+    private static final byte NUMBER_OF_WHEELS = 4;
+
+    String carBrand;
+    Engine engine;
+    Wheel[] wheels;
+    float fuelLevel = 1.0f;
+
+    public Car(String carBrand, Engine engine, Wheel[] wheels) throws CarLogicalException {
+        if (carBrand != null && engine != null &&  wheels != null) {
+            this.carBrand = carBrand;
+            this.engine = engine;
+            this.wheels = wheels;
+        } else {
+            throw new CarLogicalException("Wrong given parameters. Can't create a car!");
+        }
+    }
+
+    public float go() throws CarTechnicalException {
+        if (fuelLevel >= FUEL_AMOUNT_TO_ONE_RIDE) {
+            fuelLevel -= FUEL_AMOUNT_TO_ONE_RIDE;
+        } else {
+            throw new CarTechnicalException("There isn't enough fuel to going. You have to refuel!");
+        }
+        return fuelLevel;
+    }
+
+    public void refuel() {
+        try {
+            setFuelLevel(FULL_TANK);
+        } catch (CarLogicalException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void changeWheel(byte wheelNumber, String tireBrand, String tireType) throws CarLogicalException{
+        if (wheelNumber >= 0 && wheelNumber < NUMBER_OF_WHEELS) {
+            if (Wheel.isCorrectTireType(tireType)) {
+                byte radius = wheels[wheelNumber].getRadius();
+                wheels[wheelNumber] = new Wheel(tireType, tireBrand, radius);
+            } else {
+                throw new CarLogicalException("Unknown type of the tire. Can't change the wheel!");
+            }
+        } else {
+            throw new CarLogicalException("Wrong wheel number to changing. Can't change the wheel!");
+        }
+    }
+
+    public void printCarBrand() {
+        System.out.println("The car brand is " + carBrand);
+    }
+
+    public String getCarBrand() {
+        return carBrand;
+    }
```

Comment:
I saw that you have task printCarBrand in condition, but I see that you have method getCarBrand, so I think it is redundant
```suggestion
    public String getCarBrand() {
        return carBrand;
    }
```